src/triangle.rs

```diff
@@ -0,0 +1,655 @@
+    pub fn contains(&self, point: Point) -> bool {
+        point.x >= self.min_x()
+            && point.x < self.max_x()
+            && point.y >= self.min_y()
+            && point.y < self.max_y()
+    }
+
+    /// The greatest radius of a circle that is within the [`Triangle`]
+    #[inline]
+    pub fn radius(&self) -> f64 {
+        let ab = self.a.distance(self.b);
+        let bc = self.b.distance(self.c);
+        let ac = self.a.distance(self.c);
+
+        // semi perimeter
+        let s = (ab + bc + ac) / 2.0;
+
+        // Heron's formula
+        let area = (s * (s - ab) * (s - bc) * (s - ac)).sqrt();
+
+        area / s
+    }
+
+    /// Expand the triangle by a constant amount (`sizes`) in all directions
+    pub fn inflate(&self, size: impl Into<Size>) -> Self {
+        let size = size.into();
+
+        Self::new(
+            (self.a.to_vec2() + size.to_vec2()).to_point(),
+            (self.b.to_vec2() + size.to_vec2()).to_point(),
+            (self.c.to_vec2() + size.to_vec2()).to_point(),
+        )
+    }
+
+    /// A new [`Triangle`] with each vertecie's ordinates rounded to the nearest integer
+    #[inline]
+    pub fn round(self) -> Self {
+        Self::new(self.a.round(), self.b.round(), self.c.round())
+    }
+
+    /// A new [`Triangle`] with each vertecie's ordinates rounded up to the nearest integer
+    #[inline]
+    pub fn ceil(self) -> Self {
+        Self::new(self.a.ceil(), self.b.ceil(), self.c.ceil())
+    }
+
+    /// A new [`Triangle`] with each vertecie's ordinates rounded down to the nearest integer
+    #[inline]
+    pub fn floor(self) -> Self {
+        Self::new(self.a.floor(), self.b.floor(), self.c.floor())
+    }
+
+    /// A new [`Triangle`],
+    /// with each coordinate value rounded towards the center of the [`Triangle`]
+    /// to the nearest integer, unless they are already an integer.
+    /// That is to say this function will return the biggest possible [`Triangle`]
+    /// with integer coordinates that is a subset of `self`.
+    #[inline]
+    pub fn expand(self) -> Self {
+        Self::new(self.a.expand(), self.b.expand(), self.c.expand())
+    }
+
+    /// Returns a new [`Triangle`],
+    /// with each coordinate value rounded towards the center of the [`Triangle`]
+    /// to the nearest integer, unless they are already an integer.
+    /// That is to say this function will return the biggest possible [`Triangle`]
+    /// with integer coordinates that is a subset of `self`.
+    #[inline]
+    pub fn trunc(self) -> Self {
+        Self::new(self.a.trunc(), self.b.trunc(), self.c.trunc())
+    }
+
+    /// Scales the [`Triangle`] by a `factor`
+    #[inline]
+    pub fn scale(self, factor: f64) -> Self {
+        Self::new(
+            (self.a.to_vec2() * factor).to_point(),
+            (self.b.to_vec2() * factor).to_point(),
+            (self.c.to_vec2() * factor).to_point(),
+        )
+    }
+
+    /// Returns the [`Ellipse`] that is bounded by this [`Triangle`].
```

Comment:
As mentioned elsewhere, I think incircle and circumcircle are the methods that should be included.